### PR TITLE
Ground work for payments

### PR DESCRIPTION
**Pull request by grumbach**
- Adds public key and Commitment to dbc API

### CODE REVIEW

File: src/amount_secrets.rs [hunk 1]
```diff
@@ -45,6 +46,11 @@ impl AmountSecrets {
         self.0.blinding
     }
 
+    /// commitment
```
Comment (oetyng) [inline src/amount_secrets.rs:49]: Comments with just saying the name of a fn is not doing much :)

File: src/dbc.rs [hunk 2]
```diff
@@ -66,9 +66,15 @@ use crate::{
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct Dbc {
+    /// Encrypted information for and about this DBC's owner
```
Comment (oetyng) [inline src/dbc.rs:69]: Great that you filled in these missing comments!

Definitely helps to have the main struct members described!

File: src/dbc.rs [hunk 3]
```diff
@@ -66,9 +66,15 @@ use crate::{
...
     pub content: DbcContent,
+    /// The public key (owner) of this DBC, used for SpentProofs
+    pub public_key: PublicKey,
```
Comment (oetyng) [inline src/dbc.rs:72]: Is this not accessible through the `DbcContent`?

File: src/dbc.rs [hunk 4]
```diff
@@ -66,9 +66,15 @@ use crate::{
...
+    /// The transaction where this DBC was created
     pub transaction: DbcTransaction,
+    /// The transaction's input's SpentProofs
     pub spent_proofs: BTreeSet<SpentProof>,
```
Comment (oetyng) [inline src/dbc.rs:76]: I would even consider renaming these to:
`input_spent_proofs` and `input_spent_txs` for additional clarity.

File: src/dbc.rs [hunk 5]
```diff
@@ -390,7 +411,37 @@ pub(crate) mod tests {
 
     #[test]
     fn from_hex_should_deserialize_a_hex_encoded_string_to_a_dbc() -> Result<(), Error> {
-        let dbc = Dbc::from_hex(DBC_WITH_1_530_000_000)?;
```
Comment (oetyng) [inline src/dbc.rs:393]: Is this replacement really necessary?
Cant the public key just be gotten from this preprocessed dbc's dbc content?

File: src/dbc.rs (hunk 3 above)
Comment (grumbach) [reply src/dbc.rs:72]: Only accessible by the DBC owner through the encrypted derivation index (as it used to be a secret). 
This whole thing should be reworked and maybe we should drop that part altogether later. 
But for now, making it public is important for usability.

File: src/dbc.rs (hunk 5 above)
Comment (grumbach) [reply src/dbc.rs:393]: Not necessary. But this will save us 15 minutes every time we change the DBC struct again.

**Review by oetyng — APPROVED** [review]

**Pull request merged**